ui/lesson/HashChallenge/HashChallenge.tsx

```diff
@@ -0,0 +1,151 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import { Lesson, LessonTabs, LessonPrompt, StatusBar, Tooltip } from 'ui'
+import LessonHasher from './HashPrompt'
+import clsx from 'clsx'
+
+/**
+ * @answer {string} correct answer to the challenge problem
+ * @next {string} link to next part of chapter
+ * @label {string} label on top of input field
+ * @pattern {RegEx} pattern for input field
+ * @hints {boolean} determine whether the input field displays hints for the user
+ */
+export default function HashChallenge({
+  // children,
+  answer,
+  answerHint,
+  next,
+  label,
+  label2,
+  auto,
+  hints,
+}: {
+  // children: any
+  answer: string | number
+  next: any
+  label: string
+  label2: string
```

Comment:
Let's rename to something other than `label2` which is more descriptive of what it does, as a rule of thumb ask yourself how others are supposed to know as someone who has never used this component what `label` and `label2` do?